src/link-manager.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Object<!Array<{
+ *    target: genotet.View,
+ *    response: string
+ *  }>>>}
+ */
+genotet.linkManager.links = {};
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+
+/**
+ * Settings for link views.
+ * This part will be extended to user self-defined in next version.
+ * @const {!Array<!Object<{
+ *    source: string,
+ *    action: string,
+ *    target: string,
+ *    response: string
+ *  }>>}
+ * */
+genotet.linkManager.LINK_SETTINGS = [
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My Expression Matrix',
+    response: 'addProfile'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My Expression Matrix',
+    response: 'addProfile'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My Genome Browser',
+    response: 'locus'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My Genome Browser',
+    response: 'locus'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'nodeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'locus'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'updateTrack'
+  },
+  {
+    source: 'My Network',
+    action: 'edgeClick',
+    target: 'My 3 Tracks Genome Browser',
+    response: 'locus'
+  }
+];
+
+/** @const {string} */
+genotet.linkManager.DEFAULT_MAPPING_FILENAME = 'Direct Mapping';
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.linkManager.links = {};
+  genotet.data.mappingFile['gene-binding'] =
+    genotet.linkManager.DEFAULT_MAPPING_FILENAME;
+};
+
+/**
+ * Set up the link manager.
+ * @private
+ */
+genotet.linkManager.link_ = function() {
```

Comment:
You didn't. But you need to setup event listeners whenever new links are created. Now this link_() function sets up the listeners for every link. If an edge is just added you will only need to add listener for that specific edge. That's what "incremental" means.

You can move listener settings to register() function, so that register() creates the edge (pushing it to adjacency list) and creates event listener at the same time. Besides, you can also create unregister() to remove specific edges and event listeners.